gpustack/worker/backends/base.py

```diff
@@ -148,39 +148,55 @@ def _update_model_instance(self, id: int, **kwargs):
 
         self._clientset.model_instances.update(id=id, model_update=mi)
 
-    def get_inference_running_env(self, env: Dict[str, str] = None) -> Dict[str, str]:
-        if env is None:
-            env = os.environ.copy()
-
-        system = platform.system()
-        gpu_indexes = self._model_instance.gpu_indexes
-        if (system == "linux" or system == "windows") and gpu_indexes:
-            vendor = None
-            gpu_devices = None
-            if self._worker and self._worker.status.gpu_devices:
-                gpu_devices = self._worker.status.gpu_devices
-
-            if gpu_devices:
-                # Now use the first GPU index to get the vendor
-                first_index = gpu_indexes[0]
-                gpu_device = next(
-                    (d for d in gpu_devices if d.index == first_index), None
-                )
-                vendor = gpu_device.vendor if gpu_device else None
+    def _get_configured_env(self) -> Dict[str, str]:
+        """
+        Get the environment variables for the model instance.
+        Merge the model's env with the system env.
+        If there are conflicts, the model's env takes precedence.
 
-            # Keep GPU indexes in ascending order,
-            # this is important for some vendors like Ascend CANN,
-            # see https://github.com/gpustack/gpustack/issues/2527.
-            env_name = get_visible_devices_env_name(vendor)
-            env[env_name] = ",".join(map(str, sorted(gpu_indexes)))
+        Returns:
+            A dictionary of environment variables for the model instance.
+        """
 
-            if get_backend(self._model) == BackendEnum.VLLM:
-                set_vllm_env(env, vendor, gpu_indexes, gpu_devices)
+        env = {}
+        if not runtime_envs.GPUSTACK_RUNTIME_DEPLOY_MIRRORED_DEPLOYMENT:
```

Comment:
during mirrored deployment, gpustack-runtime(worker) can mutate the env of runner(engine), so we skip merging os environ here, trace logic from https://github.com/gpustack/runtime/blob/15026c0ad1be7edb8b7991355e39b2a15edbd3e0/gpustack_runtime/deployer/docker.py#L242C5-L242C27.